test/converters/xcresult3/converter.go

```diff
@@ -99,26 +99,38 @@ func (c *Converter) XML() (junit.XML, error) {
 		return junit.XML{}, err
 	}
 
+	useLegacyFlag := false
+
 	if supportsNewMethod {
 		log.Infof("Using new extraction method")
 
-		return parse(c.xcresultPth)
+		junitXml, err := parse(c.xcresultPth)
+		if err == nil {
+			return junitXml, nil
+		}
+
+		log.Warnf(fmt.Sprintf("Failed to parse extraction method: %s", err))
+		log.Warnf("Falling back to legacy extraction method")
+
+		logWarn("xcresult3-parsing", nil, "error: %s", err)
```

Comment:
As far as I can see, this is the only place where we use `logWarn`, and the `data` parameter is nil here. Do we need this function input?

Reply:
Also, for the first read `logWarn` doesn't really tell that it sends analytic logs. What about renaming it to `sendAnalytics` or `sendRemoteLog` or `sendRemoteWarning`?